Project/TankSim.GameHost.Unity/Assets/Scripts/Handlers/LobbyHandler.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class LobbyHandler : MonoBehaviour
+{
+
+    public Text lobbyCodeText;
+    public Button startGameButton;
+    public GameObject playersPanel;
+    public GameObject playerPanelPrefab;
+
+    private Text[] _playersTexts;
+    private bool _hasSetLobbyCode = false;
+    private GameController _gameController;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        _gameController = FindObjectOfType<GameController>();
+        if (_gameController == null)
+        {
+            Debug.LogError("GameController is not here but should be");
+            return;
+        }
+        LoadLobbyUI(_gameController.ExpectedPlayerCount);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
```

Comment:
invert if clause to avoid large if block.
if(_hasSetLobbyCode)
   UpdateLobbyUI()
   return;